core/new-gui/src/app/workspace/service/drag-drop/drag-drop.service.ts

```diff
@@ -104,9 +115,25 @@ export class DragDropService {
    *  - operatorType - the type of the operator dropped
    *  - offset - the x and y point where the operator is dropped (relative to document root)
    */
+
   public getOperatorDropStream(): Observable<{ operatorType: string, offset: Point }> {
     return this.operatorDroppedSubject.asObservable();
   }
+  public SetOffsetX(x: number) {
+    this.dragOffsetX = x;
+  }
+
+  public SetOffsetY(y: number) {
+    this.dragOffsetY = y;
+  }
+
+  public SetZoomX(x: number) {
+      this.zoomOffsetX = x;
+  }
+
+  public SetZoomY(y: number) {
+    this.zoomOffsetY = y;
+  }
```

Comment:
for these methods, rename the parameters to be `newOffsetX`, `newOffsetY`, `newZoomRatioX`, `newZoomRatioY`